src/artifact.c

```diff
@@ -5687,15 +5687,13 @@ arti_invoke(obj)
 				int role_skill;
 				struct obj *pseudo;
 				coord cc;
+				pseudo = mksobj(SPE_FIREBALL, MKOBJ_NOINIT); // create here for the skill check
```

Comment:
(Just commenting on the first time it happens)
`mksobj()` is being called, but the object is making is never being deleted. Yeah the original Tobiume was bad about this too, but could you clean this up while you're here? I think it just needs a `delobj()` at the end after each one is used. It may also be convenient to call `mksobj()` less.